tests/cases/fourslash/codeFixInferFromUsageNumberIndexSignature.ts

```diff
@@ -0,0 +1,8 @@
+/// <reference path='fourslash.ts' />
+
+// @noImplicitAny: true
+////function f([|a |]) {
+////    return a[0] + 1;
+////}
+
+verify.rangeAfterCodeFix("a: { [x: number]: number; }",/*includeWhiteSpace*/ undefined, /*errorCode*/ undefined, 0);

```

Comment:
Have we considered just inferring `Array`? Seems like that would be accurate 99% of the time.